src/main/java/org/spongepowered/api/entity/living/ArrowAttacker.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.living;
+
+import org.spongepowered.api.entity.ai.Goal;
+import org.spongepowered.api.entity.ai.task.builtin.creature.arrowattacker.AttackWithArrowAITask;
+import org.spongepowered.api.entity.projectile.source.ProjectileSource;
+
+/**
+ * {@link Agent}s which could have {@link AttackWithArrowAITask} in its {@link Goal}s.
+ */
+public interface ArrowAttacker extends Agent, ProjectileSource {
```

Comment:
So what should it be called? In vanilla this interface is called `net.minecraft.entity.IRangedAttackMob`, but some `ProjectileSource`s such as `Blaze` and `Guardian` do not implement this interface. The only method in this interface is only used by the AI in MCP code.